src/nupic/regions/TMRegion.cpp

```diff
@@ -248,12 +239,9 @@ void TMRegion::compute() {
   out = getOutput("bottomUpOut");
   if (out && (out->hasOutgoingLinks() || LogItem::isDebug())) {
     SDR& sdr = out->getData().getSDR();
-    if (args_.orColumnOutputs) { //aggregate to columns
-      tm_->getActiveCells(sdr);
-      SDR cols = tm_->cellsToColumns(sdr);
-      sdr.setSparse(cols.getSparse());
-    } else { //output as cells
-      tm_->getActiveCells(sdr);
+    tm_->getActiveCells(sdr); //active cells
+    if (args_.orColumnOutputs) { //output as columns
+      sdr = tm_->cellsToColumns(sdr);
```

Comment:
Should we add another output to TMRegion to provide access to anomaly scores, etc?

Reply:
We can decide this later when discussing "anomaly region", but yes, anomaly score would be moved to TM class, so it'll end up as a new part of existing TMRegion.

Reply:
@dkeeney if you have time, would you mind pushing a commit that adds a field "float anomalyScore" to TMRegion, and is computed from `TM.getAnomalyScore()`, please? 
I could do it but Regions are still not my cup of coffee.

Reply:
Understand.  Yes, I can do this.
But rather than a field I suggest that it be a new output. That way, the Anomaly score can be passed to some other module after each iteration.  It would be cool if we had a plot region to send it to.

If we just make it a field that can be queried then the user needs to add a callback to get it at each iteration.  We can provide that as well but it would not be as useful.

I will take a look at the old .py code to see what the old python Anomaly Region did.  I assume there was one.

Reply:
Thank you! 

> But rather than a field I suggest that it be a new output. That way, the Anomaly score can be passed to some other module after each iteration. 

Ok, this is the better solution. Would it be a problem that the output is different? (1 integer, instead of TM::getColumnDimensions(), or TM::getColumnDims + cellsPerColumn? 

> I will take a look at the old .py code to see what the old python Anomaly Region did. I assume there was one.

there wasn't any AnomalyRegion